DocumentsFromSnapshotMigration/src/main/java/com/rfs/RfsMigrateDocuments.java

```diff
@@ -130,8 +132,19 @@ public static void main(String[] args) throws Exception {
             SnapshotShardUnpacker.Factory unpackerFactory = new SnapshotShardUnpacker.Factory(repoAccessor,
                     luceneDirPath, ElasticsearchConstants_ES_7_10.BUFFER_SIZE_IN_BYTES);
 
-            run(LuceneDocumentsReader::new, reindexer, workCoordinator, processManager, indexMetadataFactory,
-                    arguments.snapshotName, arguments.indexAllowlist, shardMetadataFactory, unpackerFactory, arguments.maxShardSizeBytes);
+            try {
+                run(LuceneDocumentsReader::new, reindexer, workCoordinator, processManager, indexMetadataFactory,
+                        arguments.snapshotName, arguments.indexAllowlist, shardMetadataFactory, unpackerFactory, arguments.maxShardSizeBytes);
+            } catch (NoWorkLeftException e) {
```

Comment:
This seems like a new class of exception that has a status code associated with it.  What do you think of changing the way the these special exceptions are declared to include a status code so they don't need to be declared here?

Reply:
I think that you're talking about the exit code.  Generally, that should be a concern of the main() function and nobody else.  What happens if I have an exception set a status code, but a different exception beats it to the punch in main?  What if an exception is used by a couple different main methods/classes?  Exit code contracts should be maintained by the main classes themselves so that the responsibility doesn't get littered through the codebase